src/article.ts

```diff
@@ -100,6 +101,7 @@ export async function article(id: string, redirect: boolean, env: Env): Promise<
 		title: data.title,
 		url: new URL(id, `https://zhuanlan.zhihu.com/p/`).href,
 		content: fixImagesAndLinks(data.content),
+		reference: extractReference(data.content),
```

Comment:
why is it missed by the `content` property?

Reply:
Because it's rendered dynamically in zhihu.com. Here is the related script:

```javascript
    33846: function(er, eo, ei) {
        "use strict";
        ei.d(eo, {
            AV: function() {
                return e_
            },
            Pq: function() {
                return ew
            }
        });
        var ec = ei(37831)
          , eu = ei(64099)
          , ef = ei(87363)
          , ed = ei(95769)
          , ep = ei.n(ed)
          , eh = ei(82083)
          , em = ei(65733)
          , eg = "abcdefghijklmnopqrstuvwxyz"
          , ey = function(er) {
            var eo = er.referenceList
              , ei = er.onClick
              , ec = er.listRef;
            return (0,
            eh.jsxs)(ef.Fragment, {
                children: [(0,
                eh.jsx)("h2", {
                    children: "参考"
                }), (0,
                eh.jsx)("ol", {
                    className: "ReferenceList",
                    onClick: ei,
                    ref: ec,
                    children: ep()(eo, "refKey").map(function(er) {
                        var ei = eo.filter(function(eo) {
                            return eo.refKey === er.refKey
                        })
                          , ec = ei.length > 1 ? (0,
                        eh.jsxs)(eh.Fragment, {
                            children: [(0,
                            eh.jsx)("span", {
                                className: "ReferenceList-backHint",
                                "aria-label": "back",
                                children: "^"
                            }), ei.map(function(er, eo) {
                                var ei = "".concat(er.refKey, "_").concat(eo);
                                return (0,
                                eh.jsx)("sup", {
                                    className: "ReferenceList-backLink",
                                    children: (0,
                                    eh.jsx)("a", {
                                        href: "#".concat(ei),
                                        "data-reference-link": !0,
                                        children: eg[eo % 26]
                                    })
                                }, ei)
                            })]
                        }) : (0,
                        eh.jsx)("a", {
                            className: "ReferenceList-backLink",
                            href: "#".concat(er.refKey, "_0"),
                            "aria-label": "back",
                            "data-reference-link": !0,
                            children: "^"
                        });
                        return (0,
                        eh.jsxs)("li", {
                            id: er.refKey,
                            tabIndex: 0,
                            children: [ec, er.text && (0,
                            eh.jsx)("span", {
                                children: er.text
                            }), er.text && er.url && "\xa0", er.url && (0,
                            eh.jsx)("a", {
                                href: er.url,
                                className: "external",
                                target: "_blank",
                                rel: "noopener noreferrer",
                                children: er.url
                            })]
                        }, er.refKey)
                    })
                })]
            })
        }
          , eb = /<sup\b([^>]*data-draft-type="reference"[^>]*)>(.*?)<\/sup>/g
          , eA = function(er, eo) {
            if (!er)
                return er;
            var ei = er
              , ef = [];
            return (ei = ei.replace(eb, function(er, ei) {
                var ed = (0,
                eu.fN)(ei, {
                    decodeValue: !0
                })
                  , ep = ed["data-text"]
                  , em = void 0 === ep ? "" : ep
                  , eg = ed["data-url"]
                  , ey = void 0 === eg ? "" : eg
                  , eb = ed["data-numero"]
                  , eA = eo ? "ref_".concat(eo, "_").concat(eb) : "ref_".concat(eb)
                  , ew = ef.filter(function(er) {
                    return er.refKey === eA
                })
                  , eC = "".concat(eA, "_").concat(ew.length);
                ef.push({
                    refKey: eA,
                    text: em,
                    url: ey,
                    numero: eb
                });
                var e_ = [em, ey].filter(Boolean).join(" ");
                return (0,
                ec.renderToStaticMarkup)((0,
                eh.jsx)("sup", {
                    "data-text": em,
                    "data-url": ey,
                    "data-numero": eb,
                    "data-draft-node": "inline",
                    "data-draft-type": "reference",
                    "data-tooltip": e_,
                    "data-tooltip-preset": "white",
                    "data-tooltip-classname": "ztext-referene-tooltip",
                    children: (0,
                    eh.jsxs)("a", {
                        id: eC,
                        href: "#".concat(eA),
                        "data-reference-link": !0,
                        "aria-labelledby": eA,
                        children: ["[", eb, "]"]
                    })
                }))
            }),
            ef.length) ? ei = ei.concat((0,
            ec.renderToStaticMarkup)((0,
            eh.jsx)(ey, {
                referenceList: ef
            }))) : ei
        }
          , ew = function(er) {
            return function(eo) {
                return eA(eo, er)
            }
        }
          , eC = function(er, eo) {
            var ei = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] : {};
            if (!er.defaultPrevented && er.target.matches("a[data-reference-link]")) {
                var ec = er.target.getAttribute("href")
                  , ef = eo.querySelector(ec);
                if (ef) {
                    if (er.preventDefault(),
                    ei.onReferenceClick && er.target.hasAttribute("aria-labelledby")) {
                        ei.onReferenceClick(er);
                        return
                    }
                    ef.focus({
                        preventScroll: !0
                    }),
                    (0,
                    eu.TS)(ef),
                    document.documentElement.setAttribute("data-focus-visible", "")
                }
            }
        }
          , e_ = function(er, eo) {
            var ei = (0,
            em.Z)(function(ei) {
                return eC(ei, er, eo)
            });
            (0,
            ef.useEffect)(function() {
                if (er)
                    return er.addEventListener("click", ei),
                    function() {
                        er.removeEventListener("click", ei)
                    }
            }, [er, ei])
        }
    },
```

Reply:
understood, thanks for the investigation.